File: content/05/code/java/20--step20-implement-maximum-length/shouty/src/main/java/shouty/Network.java
package shouty;

import java.util.ArrayList;
import java.util.List;

public class Network {
    private final List<Person> listeners = new ArrayList<Person>();
    private final int range;

    public Network(int range) {
        this.range = range;
    }

    public void subscribe(Person person) {
        listeners.add(person);
    }

    public void broadcast(String message, int shouterLocation) {
        for (Person listener : listeners) {
            if (Math.abs(listener.getLocation() - shouterLocation) <= range) {
                if (message.length() <= 180) {
                    listener.hear(message);
                }
            }
        }
    }
}

File: Shouty/Shouty/Network.cs
using System;
using System.Collections.Generic;

namespace Shouty
{
    public class Network
    {
        private readonly List<Person> listeners = new List<Person>();
        private readonly int range;

        public Network(int range)
        {
            this.range = range;
        }

        public virtual void Subscribe(Person person)
        {
            listeners.Add(person);
        }

        public virtual void Broadcast(string message, int shouterLocation)
        {
            foreach (var listener in listeners)
            {
                if (Math.Abs(listener.Location - shouterLocation) <= range)
                    if (message.Length <= 180)
                        listener.Hear(message);
            }
        }
    }
}
